### PR TITLE
WIP: UI, StandardPage: add short- and view-title

### PR DESCRIPTION
**Pull request by nhaagen**
Add page-, short- and view title to Standard Page.
Bildschirmfoto_2019-10-16_19-11-09
Bildschirmfoto_2019-10-16_19-10-47

The page title is shown in the content (here, on the metabar), while the actual title-tag is short title + view title.
[Rules are here:](https://github.com/ILIAS-eLearning/ILIAS/pull/2249/files#diff-ddc95be6d038d5279fc6125f12f14745)

### CODE REVIEW

**Review by chfsx — APPROVED** [review]
LGTM, thanks! As discussed via mail ilGlobalPageTemplate now should pass the Titles e.g. via PageContentProvider (or a new one). Thanks in advance!

**Review by Amstutz — CHANGES_REQUESTED** [review]
Hi @nhaagen ,

Ready for review? If so, all good and read for JF (due to the interface changes) with two minor details:

- [ ] Documentation: Enhance the composition description of the Standard Page somewhat and indicate that there is such a thing as the short and view title and elaborate somewhat of what their function is. To me, this is not immediately clear. When would I use it, when not etc..
- [ ] Example: If possible, expand the example somewhat and showcase those two new cool features of the standard page somewhat.

Greetings
Timon

**Comment by Amstutz** [comment]
@klees as soon as good for you, attach the JF label.

**Comment by nhaagen** [comment]
Hi @chfsx and @Amstutz ,
thanks for going through this. I'll still need some time with the GlobalScreen-parts of this, but the UI Component can be discussed in JF already; I'd leave the not-ui stuff in, though, unless you object.
Cheers, 
Nils

**Comment by matthiaskunkel** [comment]
Just a little wish from my side: could we substitute the hyphen as separator between short title and view title in the title-tag by a colon (":")? This makes the hierarchy a bit clearer, e.g.: "ILIAS E-Learning : Feature Wiki" (instead of "ILIAS e-Learning - Feature Wiki").

**Comment by nhaagen** [comment]
Hey @matthiaskunkel ,
the hyphen is part of either short- or view-title and in the responsibility of the consumer.
The template itself will also allow obmission: "<title>{SHORT_TITLE} {VIEW_TITLE}</title>".
I will change the example, though.

**Comment by klees** [comment]
@nhaagen: Lets just move this to the template. Setting "- My Wiki" as page title would be as odd as setting "ILIAS :" as short title.

**Comment by matthiaskunkel** [comment]
Jour Fixe, 21 OCT 2019: We highly appreciate this suggestion and accept the PR for trunk. Please move the separator to the template and use a colon instead of a hyphen.

**Review by chfsx — APPROVED** [review]
The GS-part looks good to me, thanks!

File: src/GlobalScreen/Scope/Layout/Collector/MainLayoutCollector.php [hunk 1]
```diff
@@ -133,6 +141,19 @@ public function getFinalPage() : Page
             // PAGE
             $page_modification = $provider->getPageBuilderDecorator($called_contexts);
             $this->replaceModification($final_page_modification, $page_modification, PageBuilderModification::class);
+
+            if(get_class($provider) !== "ILIAS\GlobalScreen\Provider\GSModificationProvider") { //TODO: raus
```
Comment (nhaagen) [inline src/GlobalScreen/Scope/Layout/Collector/MainLayoutCollector.php]: @chfsx , do you have a good idea how to get rid of of this?
Id rather not extend interfaces/base classes...

Comment (chfsx) [reply src/GlobalScreen/Scope/Layout/Collector/MainLayoutCollector.php]: I still don't quite understand why this if query is needed here. Will you enlighten me please? :-)

Comment (nhaagen) [reply src/GlobalScreen/Scope/Layout/Collector/MainLayoutCollector.php]: I now extended ModificationProvider-interface; works like a charme ;)

**Comment by klees** [comment]
@nhaagen This is ready for final review, right?

**Comment by nhaagen** [comment]
@klees Yes, it is.

**Review by klees — APPROVED** [review]

**Comment by klees** [comment]
@chfsx This can be merged from your perspective, right? The UI Framework side looks fine (and mostly trivial).

**Comment by chfsx** [comment]
@klees  yes this LGTM!

**Pull request merged**